vpr/src/analytical_place/partial_placement.cpp

```diff
@@ -38,23 +39,23 @@ double PartialPlacement::estimate_post_placement_wirelength(const APNetlist& net
     // we want to estimate the post-placement wirelength, we do not want the
     // flat placement positions of the blocks. Instead we compute the HPWL over
     // the tiles that the flat placement is placing the blocks over.
-    unsigned total_hpwl = 0;
+    double total_hpwl = 0;
     for (APNetId net_id : netlist.nets()) {
-        // Note: Other wirelength estimates in VTR ignore global nets; however
-        //       it is not known if a net is global or not until packing is
-        //       complete. For now, we just approximate post-placement wirelength
-        //       using the HPWL (in tile space).
-        // TODO: The reason we do not know what nets are ignored / global is
-        //       because the pin on the tile that the net connects to is what
-        //       decides if a net is global / ignored for place and route. Since
-        //       we have not packed anything yet, we do not know what pin each
-        //       net will go to; however, we can probably get a good idea based
-        //       on some properties of the net and the tile its going to / from.
-        //       Should investigate this to get a better estimate of wirelength.
-        double min_x = std::numeric_limits<unsigned>::max();
-        double max_x = std::numeric_limits<unsigned>::lowest();
-        double min_y = std::numeric_limits<unsigned>::max();
-        double max_y = std::numeric_limits<unsigned>::lowest();
+        // To align with other wirelength estimators in VTR (for example in the
+        // placer), we do not include global nets (clocks, etc.) in the wirelength
+        // calculation.
+        if (netlist.net_is_global(net_id))
+            continue;
+
+        // Similar to the placer, weight the wirelength of this net as a function
+        // of its fanout.
+        // TODO: Should update the costs to 3D.
```

Comment:
We should comment that these fanouts are at the AP netlist (unclustered) level so the correction factors will lead to a somewhat higher hpwl prediction than after clustering.